### PR TITLE
IGNITE-20636 Add to the MakeIndexAvailableCommand the ability to use only the indexId

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-20636

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java [hunk 1]
```diff
@@ -367,12 +367,11 @@ public static String pkIndexName(String tableName) {
...
 
     /**
-     * Returns index with given name.
+     * Returns index descriptor.
      *
      * @param schema Schema to look up index in.
      * @param name Name of the index of interest.
-     * @return Table with given name.
-     * @throws IndexNotFoundValidationException If index with given name is not exists.
+     * @throws IndexNotFoundValidationException If index is not exists.
```
Comment (rpuch) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:374]:
```suggestion
     * @throws IndexNotFoundValidationException If index does not exist.
```

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java [hunk 2]
```diff
@@ -383,4 +382,21 @@ static CatalogIndexDescriptor indexOrThrow(CatalogSchemaDescriptor schema, Strin
 
         return index;
     }
+
+    /**
+     * Returns index descriptor.
+     *
+     * @param catalog Catalog to look up index in.
+     * @param indexId ID of the index of interest.
+     * @throws IndexNotFoundValidationException If index is not exists.
```
Comment (rpuch) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:391]:
```suggestion
     * @throws IndexNotFoundValidationException If index does not exist.
```

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/storage/MakeIndexAvailableEntry.java [hunk 3]
```diff
@@ -72,16 +73,60 @@ public CatalogEvent eventType() {
...
+
+        assert schema != null : table.schemaId();
+
+        return schema;
     }
 
-    /** Returns schema name. */
-    public String schemaName() {
-        return schemaName;
+    private static CatalogIndexDescriptor createReadWriteIndex(CatalogIndexDescriptor source, long causalityToken) {
```
Comment (rpuch) [inline modules/catalog/src/main/java/org/apache/ignite/internal/catalog/storage/MakeIndexAvailableEntry.java:95]: Again, 2 terminologies. Writing in some documentation that 'RW is same as Available' will not be enough if all around the code we see either Available, or RW. My suggestion is to pick 1 principal terminology (I suggest 'Registered/Available') and write the new code in accordance with it, and then change all the terminology in accordance with this one (for the code that was written earlier).

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java (hunk 1 above)
Comment (tkalkirill) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:374]: fix it

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java (hunk 2 above)
Comment (tkalkirill) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java:391]: fix it

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/storage/MakeIndexAvailableEntry.java (hunk 3 above)
Comment (tkalkirill) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/storage/MakeIndexAvailableEntry.java]: Tried renaming everything related to this problem.

**Review by rpuch — APPROVED** [review]

**Pull request merged**